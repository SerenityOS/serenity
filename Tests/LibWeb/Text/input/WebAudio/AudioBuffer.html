<script src="../include.js"></script>
<script>
    test(() => {
        // Invalid constructors
        const invalidOptions = [
            { numberOfChannels: 0, length: 17, sampleRate: 10_002 }, // 0 channels (below min)
            { numberOfChannels: 33, length: 17, sampleRate: 10_002 }, // 33 channels (above max)
            { numberOfChannels: 3, length: 0, sampleRate: 10_002 }, // 0 length
            { numberOfChannels: 3, length: 17, sampleRate: 7999 }, // 7999 sample rate (below min)
            { numberOfChannels: 3, length: 17, sampleRate: 192001 }, // 192001 sample rate (above max)
        ];

        for (let invalidOption of invalidOptions) {
            try {
                const buffer = new AudioBuffer(invalidOption);
                println(`FAIL: created buffer ${buffer}`);
            } catch (e) {
                println(`Error creating AudioBuffer: '${e}'`);
            }
        }

        // Valid Constructor
        const options = { numberOfChannels: 3, length: 17, sampleRate: 10_002 };
        const buffer = new AudioBuffer(options);

        println(buffer.numberOfChannels);
        println(buffer.length);
        println(buffer.sampleRate);

        // Check each of the channels
        for (let k = 0; k < options.numberOfChannels; ++k) {
            const data = buffer.getChannelData(k);
            println(`Got ${data.constructor.name}, length = ${data.length}`);
            println(data.length);

            const dataAgain = buffer.getChannelData(k);
            println(`Data equals itself: ${data === dataAgain}`);
        }

        // Out of range channel
        try {
            buffer.getChannelData(options.numberOfChannels);
            println("FAIL: No exception thrown");
        } catch (e) {
            println(`Error getting channel data: '${e}'`);
        }
    });
</script>
