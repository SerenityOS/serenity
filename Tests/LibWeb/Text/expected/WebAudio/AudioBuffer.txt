Error creating AudioBuffer: 'NotSupportedError: Number of channels must not be '0''
Error creating AudioBuffer: 'NotSupportedError: Number of channels is greater than allowed range'
Error creating AudioBuffer: 'NotSupportedError: Length of buffer must be at least 1'
Error creating AudioBuffer: 'NotSupportedError: Sample rate is outside of allowed range'
Error creating AudioBuffer: 'NotSupportedError: Sample rate is outside of allowed range'
3
17
10002
Got Float32Array, length = 17
17
Data equals itself: true
Got Float32Array, length = 17
17
Data equals itself: true
Got Float32Array, length = 17
17
Data equals itself: true
Error getting channel data: 'IndexSizeError: Channel index is out of range'
